### ISSUE TITLE
Trim empty lines at the top of shader program

### ISSUE THREAD

**Issue by campadrenalin** [Dav1dde/glamour#24]
If you have blank lines before the start of the directives, none of the directives will be stored. This can not just be surprising, but the default shader versions can lull you into a false sense of security that the directives are working... until you need to bump the version. And it's especially bad if you start from the [example code in this repository](https://github.com/Dav1dde/glamour/blob/master/example/example.d#L42), which has exactly this pitfall in it (and is also a great example of how blank lines can end up at the top of a shader source in real code).

Basically, I just burned half an hour tracking down why I was getting compile failures about needing a version >= 140, after bumping the version to 330. That's bad, but fixable, so let's do that!

**Comment by Dav1dde** [comment]
Good catch! Thanks for the report.
